peeringmod.py

```diff
@@ -0,0 +1,233 @@
+import json
+import os
+import subprocess
+import sys
+from collections import defaultdict
+from dataclasses import asdict, dataclass
+from ipaddress import IPv4Network, IPv6Network
+from typing import List
+from wsgiref import headers
+
+import jinja2
+import jsonschema
+import requests
+
+
+@dataclass
+class AnnouncementGenerator:
+    prefix: IPv4Network or IPv6Network
+    muxes: List[str]
+    details: List[dict]
+
+    def __init__(self, prefix, muxes, **kwargs):
+        self.prefix = prefix
+        self.muxes = muxes
+        self.details = list()
+
+        self.add_details(muxes, **kwargs)
+
+    def add_details(self, muxes, **kwargs):
```

Comment:
```{python}
dictdetails = dict(kwargs)
dictdetails["muxes"] = muxes
self.details.append(dictdetails)
```

But even then I am not a fan of this design.  It seems like all of the logic is bundled in the `details` field.  I'd prefer if we made things more explicit.  I mean, how are the different details used?